code/game/objects/items/tools/spess_knife.dm

```diff
@@ -0,0 +1,152 @@
+		effectiveness = 100, \
+		disabled = TRUE, \
+	)
+	options = list(
+		NO_TOOL = image(icon = 'icons/obj/tools.dmi', icon_state = initial(icon_state)),
+		TOOL_KNIFE = image(icon = 'icons/obj/tools.dmi', icon_state = "[initial(icon_state)]_[TOOL_KNIFE]"),
+		TOOL_SCREWDRIVER = image(icon = 'icons/obj/tools.dmi', icon_state = "[initial(icon_state)]_[TOOL_SCREWDRIVER]"),
+		TOOL_WIRECUTTER = image(icon = 'icons/obj/tools.dmi', icon_state = "[initial(icon_state)]_[TOOL_WIRECUTTER]"),
+	)
+
+/obj/item/spess_knife/attack_self(mob/user, modifiers)
+	. = ..()
+	var/old_tool_behaviour = tool_behaviour
+	var/new_tool_behaviour = show_radial_menu(user, src, options, require_near = TRUE, tooltips = TRUE)
+	if(isnull(new_tool_behaviour) || new_tool_behaviour == tool_behaviour)
+		return
+	if(new_tool_behaviour == NO_TOOL)
+		tool_behaviour = null
+	else
+		tool_behaviour = new_tool_behaviour
+
+	var/mistake_chance = HAS_TRAIT(user, TRAIT_CLUMSY) ? wrong_tool_prob * 5 : wrong_tool_prob
+	var/mistake_occured = FALSE
+	if(!isnull(tool_behaviour) && prob(mistake_chance))
+		do
+			pick_tool() // Pick random tool, excluding the desired and current one
+		while (tool_behaviour == new_tool_behaviour || tool_behaviour == old_tool_behaviour)
+		mistake_occured = TRUE
+
+	if(isnull(tool_behaviour))
+		w_class = WEIGHT_CLASS_TINY
+		balloon_alert(user, "folded")
+	else
+		w_class = WEIGHT_CLASS_SMALL
+		balloon_alert(user, mistake_occured ? "oops! [tool_behaviour] out" : "[tool_behaviour] out")
+
+	update_tool_parameters()
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/weapons/empty.ogg', 50, TRUE)
+
+/// Used to pick random tool behavior for the knife
+/obj/item/spess_knife/proc/pick_tool()
+	tool_behaviour = pick_weight(list(
+		TOOL_KNIFE = 10,
+		TOOL_SCREWDRIVER = 10,
+		TOOL_WIRECUTTER = 10,
+		TOOL_WRENCH = 5,
+		TOOL_SHOVEL = 2,
+		TOOL_SAW = 2,
+		TOOL_ROLLINGPIN = 1,
+	))
+
+/// Used to update sounds and tool parameters during switching
+/obj/item/spess_knife/proc/update_tool_parameters()
+	var/datum/component/butchering/butchering = src.GetComponent(/datum/component/butchering)
+	butchering.butchering_enabled = tool_behaviour == TOOL_KNIFE
+	RemoveElement(/datum/element/eyestab)
+	switch(tool_behaviour)
+		if(TOOL_KNIFE)
+			force = 8
+			sharpness = SHARP_EDGED
+			hitsound = 'sound/weapons/bladeslice.ogg'
+			usesound = initial(usesound)
+			AddElement(/datum/element/eyestab)
+		if(TOOL_SCREWDRIVER)
+			force = 4
+			sharpness = SHARP_POINTY
+			hitsound = 'sound/weapons/bladeslice.ogg'
+			usesound = list('sound/items/screwdriver.ogg', 'sound/items/screwdriver2.ogg')
+			AddElement(/datum/element/eyestab)
+		if(TOOL_WIRECUTTER)
+			force = 4
+			sharpness = NONE
+			hitsound = 'sound/items/wirecutter.ogg'
+			usesound = 'sound/items/wirecutter.ogg'
+		if(TOOL_WRENCH)
+			force = 4
+			sharpness = NONE
+			hitsound = initial(hitsound)
+			usesound = 'sound/items/ratchet.ogg'
+		if(TOOL_SHOVEL)
+			force = 6
+			sharpness = SHARP_EDGED
```

Comment:
Shovels are sharp?

Reply:
They are.

Reply:
honestly i don't like this incase something changes in the future.
let's get the values we change from the typepaths instead.

```
var/obj/item/shovel/shovel_obj
sharpness = initial(shovel_obj.sharpness)
usesound = initial(shovel_obj.usesound)
```

do that for all the tool types. that'll ensure everything is always correct.